engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/HashMapGetNode.java

```diff
@@ -37,19 +40,18 @@ public static HashMapGetNode build() {
 
   @Specialization(guards = "interop.hasHashEntries(self)", limit = "3")
   Object hashMapGet(
-      VirtualFrame frame,
-      State state, Object self, Object key, Object defaultValue,
-      @CachedLibrary("self") InteropLibrary interop,
-      @Shared @Cached("build()") ThunkExecutorNode thunkExecutorNode) {
-    if (interop.isHashEntryReadable(self, key)) {
+    VirtualFrame frame,
+    State state, Object self, Object key, Object defaultValue,
+    @CachedLibrary("self") InteropLibrary interop,
+    @Shared @Cached("build()") ThunkExecutorNode thunkExecutorNode
+  ) {
       try {
         return interop.readHashValue(self, key);
-      } catch (UnsupportedMessageException | UnknownKeyException e) {
-        throw new IllegalStateException(e);
+      } catch (UnknownKeyException e) {
+        return thunkExecutorNode.executeThunk(frame, defaultValue, state, TailStatus.NOT_TAIL);
```

Comment:
Why dropping the condition check in favour of catching an exception? Both are PE-friendly?

Reply:
First of all we should drop the `IllegalStateException` - we shouldn't crash Enso interpreter (which `ISE` does) just because some random object from polyglot world is inconsistent.

`UnknownKeyException` is fast exception - e.g. no slowdown then obtaining it. And the code even looks simpler without the `isReadable` check.